### ISSUE TITLE
Use hidden datastreams for aggregated metrics with a non-default rollup interval 

### ISSUE THREAD

**Issue by simitt** [elastic/apm-server#10047]
Update Jan 23rd 2023:

Instead of adding a suffix to the `metricset.name`, we are moving forward with leveraging hidden datastreams for metrics with a non-default aggregation interval (other than 1minute). 
This should be applied to transaction, service and the new service summary metrics. 

Outdated: 
~According to https://github.com/elastic/kibana/issues/148507#issuecomment-1380830848, add a suffix `.internal` to the `metricset.name` for any rolled up metrics with a interval different than the `1m` default interval.~

~`.internal` is used instead of `-internal` to align with the [data stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme) because data stream name is derived from metricset.name currently~

**Comment by felixbarny** [comment]
> .internal is used instead of -internal to align with the [data stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme).

I don't have a strong opinion on the naming but I don't quite understand what this has to do with the data stream naming scheme. Aren't we just adding it to the `metricset.name` field? Or are we also planning to change the data stream name?

**Comment by carsonip** [comment]
> I don't have a strong opinion on the naming but I don't quite understand what this has to do with the data stream naming scheme. Aren't we just adding it to the `metricset.name` field? Or are we also planning to change the data stream name?

It is an implementation detail in [datastream.go:99](https://github.com/elastic/apm-server/blob/main/internal/model/modelprocessor/datastream.go#L99). We can
1. keep the implementation and change apmpackage; or
2. change the implementation in datastream.go

**Comment by simitt** [comment]
@carsonip the implementation doesn't seem to be too hard to change, but I haven't checked if it has other implications within the code. Could you do a quick review on how easy it would be to not have the suffix reflected in the datastream?

**Comment by carsonip** [comment]
We are going to utilize hidden data streams instead of modifying metricset.name to achieve the same goal. Pending changes on package spec https://github.com/elastic/package-spec/pull/465 and fleet.

**Comment by felixbarny** [comment]
All the pending changes in the package spec and in Fleet have been merged. I think we're unblocked now. Or do we need to wait for a release or a snapshot of some sort?

**Comment by carsonip** [comment]
> All the pending changes in the package spec and in Fleet have been merged. I think we're unblocked now. Or do we need to wait for a release or a snapshot of some sort?

I can re-test https://github.com/elastic/apm-server/pull/10070 and see if it works

**Comment by carsonip** [comment]
I'm getting
```
2023/01/23 09:46:26 generating integration package v8.7.0 in "build/apmpackage"
cd build/apmpackage && /home/carson/projects/apm-server/build/linux/elastic-package build
2023/01/23 09:46:28  INFO New version is available - v0.71.0. Download from: https://github.com/elastic/elastic-package/releases/tag/v0.71.0
Build the package
README.md file rendered: /home/carson/projects/apm-server/build/apmpackage/docs/README.md
2023/01/23 09:46:28  INFO License text found in "/home/carson/projects/apm-server/LICENSE.txt" will be included in package
Error: building package failed: invalid content found in built zip package: found 12 validation errors:
   1. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_10m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
   2. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_1m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
   3. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_60m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
   4. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_destination_10m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
   5. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_destination_1m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
   6. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_destination_60m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
   7. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_summary_10m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
   8. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_summary_1m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
   9. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/service_summary_60m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
  10. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/transaction_10m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
  11. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/transaction_1m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
  12. file "/home/carson/projects/apm-server/build/packages/apm-8.7.0.zip/data_stream/transaction_60m_metrics/manifest.yml" is invalid: field elasticsearch.index_template: Additional property data_stream is not allowed
```

I think we need a bump in [package-spec](https://github.com/elastic/package-spec) and [elastic-package](https://github.com/elastic/elastic-package)
